sklego/meta.py

```diff
@@ -117,3 +117,60 @@ def predict(self, X):
                            .loc[lambda d: d['new'] == 1]
                            .itertuples())
             raise ValueError(f"found a group(s) {culprits} in `.predict` that was not in `.fit`")
+
+
+class DecayEstimator(BaseEstimator):
+    """
+    Morphs an estimator suchs that the training weights can be
+    adapted to ensure that points that are far away have less weight.
+    Note that it is up to the user to sort the dataset appropriately.
+    This meta estimator will only work for estimators that have a
+    "sample_weights" argument in their `.fit()` method.
+
+    The DecayEstimator will use exponential decay to weight the parameters.
+
+    w_{t-1} = decay * w_{t}
+
+    """
+
+    def __init__(self, model, decay: float = 0.999):
+        self.model = model
+        self.decay = decay
```

Comment:
Maybe it's premature but I could imagine that passing a decay function to class rather than a single linear decay weight could add some more flexibility. Something like:

```
def linear_decay(decay):
   return lambda distance: ...

class GeneralDecay():
    def __init__(self, model, decay_func):
        ....

LinearDecay = lambda model, decay_weight: GeneralDecay(model, linear_decay(decay_weight)) 
```

Reply:
I agree on this but this requires a bit of thought. We could internally allow for multiple variants of decay but perhaps this is better implemented as a string (this might make gridsearch results a bit nicer). This might be nice to skip in this PR because I like the idea of adding a decay function as an exercise for one of our students. 

A more involved implementation (one that is interesting) involves a convenient method to assign decay not based on time/order but rather on things like: 

- is it a holiday? 
- is this customer important? 
- any other property that is in the data 

For this latter part I do not see a nice implementation.

Reply:
A more involved implementation (one that is interesting) involves a convenient method to assign decay not based on time/order but rather on things like: 

- is it a holiday? 
- is this customer important? 
- any other property that is in the data 

For this latter part I do not see a nice implementation yet though.

Reply:
Adding a decay function is indeed a nice exercise but the framework should already allow for it beforehand I would say. I'm open to skipping it for this PR though

Reply:
I've added the option for `decay_func`. It currently does nothing but it gives space. 

```
    def __init__(self, model, decay: float = 0.999, decay_func="exponential"):
        self.model = model
        self.decay = decay
        self.func = decay_func
```